src/character/gear/EditableItemEntry.tsx

```diff
@@ -0,0 +1,176 @@
+import React, { useContext, useEffect, useState } from "react";
+import styled, { css } from "styled-components";
+import { Theme } from "../../Theme";
+import { CustomItem, CustomItemData } from "../../utils/types/CustomItem";
+import { Modal, TextArea } from "react-aria-components";
+import { CharacterContext } from "../../DeadlandsCompanion";
+import { Button } from "../../shared/buttons/Button";
+import { useLongPress } from "use-long-press";
+import { deleteCustomitem, updateCustomItem } from "./CustomItemService";
+
+const EditableItemEntryWrapper = styled.div`
+  display: flex;
+  flex-direction: column;
+  justify-content: space-between;
+  border-radius: 4px;
+  ${({}) => css`
+    background-color: ${Theme.Surface[300]};
+  `}
+`;
+type EditableItemEntryProps = {
+  item: CustomItem;
+};
+
+const ModalContents = styled.div`
+  display: flex;
+  flex-direction: column;
+  gap: 8px;
+  align-items: center;
+  justify-content: center;
+`;
+
+const ItemTitle = styled.h4`
+  margin: 0;
+  padding: 0;
+  font-size: 1em;
+`;
+
+const ItemWrapper = styled.div`
+  display: flex;
+  flex-direction: column;
+  gap: 8px;
+  padding: 4px;
+  border-radius: 8px;
+  justify-content: space-between;
+`;
+
+const ButtonsWrapper = styled.div`
+  display: flex;
+  flex-direction: row;
+  gap: 8px;
+  padding: 4px;
+  border-radius: 8px;
+  justify-content: flex-end;
+  button {
+    padding: 4px;
+    border-radius: 4px;
+    border: none;
+  }
+`;
+
+const StyledInput = styled.input`
+  padding: 4px;
+  border-radius: 4px;
+  border: none;
+`;
+
+const EditableItemEntry: React.FC<EditableItemEntryProps> = ({ item }) => {
+  const characterContext = useContext(CharacterContext);
+  const [editable, setIsEditable] = useState(false);
+  const [expanded, setIsExpanded] = useState(false);
+  const [editedName, setEditedName] = useState(item.title);
+  const [editedDesc, setEditedDesc] = useState(item.description);
+  const [showDeletionModal, setShowDeletionModal] = useState(false);
+
+  useEffect(() => {
+    if (item.title !== editedName) setEditedName(item.title);
+  }, [item.title]);
```

Comment:
I hate this pattern, but I've never managed to find a cleaner alternative...

Reply:
One alternative would be to encapsulate all the state, inputs, and save/cancel buttons related to editing the item in their own conditionally rendered component and letting the conditional be the `editable` state. That way, the `editedName` state will be reset to `item.title` every time the component is put into editing mode. This also comes with the benefit that external changes to the title will not affect the user's ongoing edit.

Reply:
Good idea; I've implemented that :)